pkg/config/config_template.yaml

```diff
@@ -445,7 +451,8 @@ api_key:
 
   ## @param env - string - optional - default: none
   ## The environment tag that Traces should be tagged with.
-  ## Inherits from "env" tag if "none" is applied here.
+  ## If "none" is set here the value will be inherited first from the top level
```

Comment:
```sh
  ## If left empty or set to "none", the value will be inherited from the top level
  ## "env" config option, if set.
```
Maybe this is enough? And then, we can mention on the top level "env" option that the `env:` tag will override this setting.

Reply:
the `env:` tag will only overwrite it for trace, the agent will send both. I think we have to explain it.

Reply:
Oh, right. That’s true. Maybe we can phrase it a bit better? The current one is a bit hard to understand. Maybe @l0k0ms can suggest something as the docs expert.

Reply:
It's a bit of a weird edge case, but if I set the following in the config:

```yaml
env: "foo"
apm_config:
  env: "none"
```

based on the current documentation here, I'd expect the APM Agent to pick up `foo` as the env tag. But it I understand the code correctly, the APM Agent will actually send `none` as the env tag. So we could rephrase this to:

```
If this option is left unset or set to an empty value, the environment tag will instead be inherited from the first config option that's actually set to a non-empty value (if any) in this list:
1. top-level `env` option.
2. `env:` tag under the top-level `tags` option.
```

Reply:
I edited the description